### PR TITLE
 fix(docker): Do not clear an existing imageId even if fields cannot be found in registry

### PR DESCRIPTION
**Pull request by jrsquared**
Currently, if an imageId is present in the config on load of this control, and the image cannot be found in the registry, it will clear it and set imageId to nothing.

This is bad.

After this fix, when an image cannot be found, it will switch to manual entry with a warning.
screen shot 2019-01-09 at 8 45 45 pm

### CODE REVIEW

File: app/scripts/modules/core/src/presentation/forms/forms.less [hunk 1]
```diff
@@ -171,3 +172,36 @@
...
+    padding: 7px;
+  }
+
+  .message {
+    margin: 8px 8px 8px 0;
+    align-self: center;
+  }
+}
+.warningMessage {
+  background-color: rgba(255, 220, 60, 0.4);
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/presentation/forms/forms.less]: Can we get these colors into the style guide as color variables?

File: app/scripts/modules/docker/src/image/DockerImageAndTagSelector.tsx [hunk 2]
```diff
@@ -38,9 +38,9 @@ export interface IDockerImageAndTagSelectorProps {
 
 export interface IDockerImageAndTagSelectorState {
   accountOptions: Array<Option<string>>;
+  switchedManualWarning: string;
```
Comment (anotherchrisberry) [inline app/scripts/modules/docker/src/image/DockerImageAndTagSelector.tsx]: If this were optional, you could save a line of code and reduce our bundles sizes considerably.

File: app/scripts/modules/core/src/presentation/forms/forms.less (hunk 1 above)
Comment (jrsquared) [reply app/scripts/modules/core/src/presentation/forms/forms.less]: This was used from an in-progress form design from @gcomstock, we will get it all styleguide-ed once those designs finalize.

File: app/scripts/modules/docker/src/image/DockerImageAndTagSelector.tsx (hunk 2 above)
Comment (jrsquared) [reply app/scripts/modules/docker/src/image/DockerImageAndTagSelector.tsx]: True.

**Comment by gcomstock** [comment]
Looks great! Thanks for sharing.

**Pull request merged**